autogluon/utils/tabular/ml/models/catboost/catboost_model.py

```diff
@@ -146,6 +146,11 @@ def _fit(self, X_train, y_train, X_val=None, y_val=None, time_limit=None, **kwar
 
         params = self.params.copy()
         num_features = len(self.features)
+        if num_gpus != 0:
+            if 'task_type' not in params:
+                params['task_type'] = 'GPU'
+                # TODO: Confirm if GPU is used in HPO (Probably not)
```

Comment:
any idea why GPU isn't utilized during HPO?

Reply:
Disjoint code calls, I don't think the num_gpus param is passed with HPO. In general, I'm in favor of moving num_gpus into params instead of fit, that gives us a lot more flexibility in terms of per-model GPU usage, and keeps model .fit args clean. It could be part of AG_args_fit.